src/Drawable.js

```diff
@@ -564,12 +568,64 @@ class Drawable {
      * @return {!Rectangle} Rough axis-aligned bounding box for Drawable.
      */
     getAABB (result) {
-        if (this._transformDirty) {
-            this._calculateTransform();
+        if (this._aabbDirty) {
+            if (this._transformDirty) {
+                this._calculateTransform();
+            }
+
+            // Calculate the rotated "logical" bounding box
+            const size = this.skin.nativeSize;
+            const center = this.skin.nativeRotationCenter;
+
+            // TODO: This is a tad bit janky but saves some math. Worth it?
+            const cos = this._rotationMatrix[0];
+            const sin = this._rotationMatrix[1];
+
+            // TODO: this can likely be golfed
+            const origLeft = center[0] * this._scale[0] / 100;
+            const origTop = -center[1] * this._scale[1] / 100;
+            const origRight = (center[0] - size[0]) * this._scale[0] / 100;
+            const origBottom = (size[1] - center[1]) * this._scale[1] / 100;
+
+            const leftCos = origLeft * cos;
+            const rightCos = origRight * cos;
```

Comment:
It bothers me that top and bottom are grouped but left and right are intermingled in the ordering